src/vcTiffConvert.cpp

```diff
@@ -10,157 +10,592 @@
+  {
+    // Currently do not support 16-bit floats
+    uint64_t ind = pixelIndex * format.samplesPerPixel + sampleIndex;
+    if (format.sampleFormat == SAMPLEFORMAT_INT)
+      pSample->t_int64[sampleIndex] = ((int16_t *)(pBuf))[ind];
+    else if (format.sampleFormat == SAMPLEFORMAT_UINT)
+      pSample->t_uint64[sampleIndex] = ((uint16_t *)(pBuf))[ind];
+    else
+      UD_ERROR_SET(udE_InvalidConfiguration);
+  }
+  else if (format.bitsPerSample == 32)
+  {
+    uint64_t ind = pixelIndex * format.samplesPerPixel + sampleIndex;
+    if (format.sampleFormat == SAMPLEFORMAT_INT)
+      pSample->t_int64[sampleIndex] = ((int32_t *)(pBuf))[ind];
+    else if (format.sampleFormat == SAMPLEFORMAT_UINT)
+      pSample->t_uint64[sampleIndex] = ((uint32_t *)(pBuf))[ind];
+    else if (format.sampleFormat == SAMPLEFORMAT_IEEEFP)
+      pSample->t_double[sampleIndex] = (double)((float *)(pBuf))[ind];
+    else
+      UD_ERROR_SET(udE_InvalidConfiguration);
+  }
+  else if (format.bitsPerSample == 64)
+  {
+    uint64_t ind = pixelIndex * format.samplesPerPixel + sampleIndex;
+    if (format.sampleFormat == SAMPLEFORMAT_INT)
+      pSample->t_int64[sampleIndex] = ((int64_t *)(pBuf))[ind];
+    else if (format.sampleFormat == SAMPLEFORMAT_UINT)
+      pSample->t_uint64[sampleIndex] = ((uint64_t *)(pBuf))[ind];
+    else if (format.sampleFormat == SAMPLEFORMAT_IEEEFP)
+      pSample->t_double[sampleIndex] = ((double *)(pBuf))[ind];
+    else
+      UD_ERROR_SET(udE_InvalidConfiguration);
+  }
+
+  result = udE_Success;
+epilogue:
+  return result;
+}
 
-class TiffFullImageReader : public TiffPixelReader
+static udError vcTiff_LoadTile(vcTiffConvertData *pData, uint32_t x, uint32_t y)
 {
-public:
+  udError result;
+  uint8_t *pBuf = nullptr;
+  uint32_t px, py;
+
+  UD_ERROR_NULL(pData, udE_InvalidParameter);
+
+  px = x * pData->tile.tileDimensions.x;
+  py = y * pData->tile.tileDimensions.y;
 
-  TiffFullImageReader()
-    : TiffPixelReader()
-    , m_pRaster(nullptr)
-    , m_currentPixel(0)
-    , m_imageDimensions{}
+  pData->tile.currentPixel = 0;
+  pData->tile.currentTileCoords = {x, y};
+
+  if (pData->format.planarconfig == PLANARCONFIG_CONTIG)
+  {
+    if (pData->tile.pRaster == nullptr)
+    {
+      pData->tile.pRaster = (uint8_t *)_TIFFmalloc(TIFFTileSize(pData->pTif));
+      UD_ERROR_NULL(pData->tile.pRaster, udE_MemoryAllocationFailure);
+    }
+    UD_ERROR_IF(TIFFReadTile(pData->pTif, pData->tile.pRaster, px, py, 0, 0) == 0, udE_ReadFailure);
+  }
+  else
   {
+    if (pData->tile.pRaster == nullptr)
+    {
+      pData->tile.pRaster = (uint8_t *)_TIFFmalloc(TIFFTileSize(pData->pTif) * udMin(pData->format.samplesPerPixel, MAX_SAMPLES));
+      UD_ERROR_NULL(pData->tile.pRaster, udE_MemoryAllocationFailure);
+    }
+
+    size_t pixelCount = (size_t)pData->tile.tileDimensions.x * pData->tile.tileDimensions.y;
+    pBuf = (uint8 *)_TIFFmalloc(TIFFTileSize(pData->pTif)); // TODO could allocated this once per image
```

Comment:
Given that this function frees `pBuf`, is there any reason to use `_TIFFmalloc` over `udAllocType` here?